tempodb/encoding/vparquet/compactor.go

```diff
@@ -82,6 +80,21 @@ func (c *Compactor) Compact(ctx context.Context, l log.Logger, r backend.Reader,
 			return rows[0], nil
 		}
 
+		// Total
+		if c.opts.MaxBytesPerTrace > 0 {
```

Comment:
this ditches traces a bit more aggressively than the v2 logic. let's keep an eye on spans discarded as we roll this out.

Reply:
Yes that is true, it is comparing `sum(len(inputs))` whereas v2 compared against `len(output)`.  However, in this case the mem usage from parsing and combining a large trace is significant (possibly a few GB).  If we can think of a way to make this more efficient/precise, let's do it.